### PR TITLE
Add matomo events

### PR DESCRIPTION
**Pull request by corwintines**
## Description

- Add matomo events throughout the site

### CODE REVIEW

**Comment by minimalsm** [comment]
@corwintines is there a way to test if these are working as expected outside of prod 🤔

File: src/templates/use-cases.tsx [hunk 1]
```diff
@@ -346,22 +346,47 @@ const UseCasePage = ({
       {
         text: t("template-usecase-dropdown-defi"),
         to: "/defi/",
+        matomo: {
+          eventCategory: "use cases menuy",
```
Comment (minimalsm) [inline src/templates/use-cases.tsx:350]: Typo?

```suggestion
          eventCategory: "use cases menu",
```

**Comment by corwintines** [comment]
> @corwintines is there a way to test if these are working as expected outside of prod 🤔

Not easily, unfortunately. You can edit the matomo.ts file to work locally if you want to test locally, but don't think that there's an easy way to do this on preview deploys

File: src/templates/use-cases.tsx [hunk 2]
```diff
@@ -324,22 +324,47 @@ const UseCasePage = ({
       {
         text: t("template-usecase-dropdown-defi"),
         to: "/defi/",
+        matomo: {
+          eventCategory: "use cases menu",
+          eventAction: "click",
+          eventName: "/defi/",
```
Comment (pettinarip) [inline src/templates/use-cases.tsx:330]: Just noting that this differs from the requested eventName in the Notion card. Those don't have the `/`s. Maybe is not a problem and we can leave it like is.

```suggestion
          eventName: "defi",
```

Comment (pettinarip) [reply src/templates/use-cases.tsx:330]: Same with the other cases in this file.

**Review by pettinarip — APPROVED** [review]
LGTM!

Wanted to mention that perhaps for other iterations we could create a new data structure for events + its type to avoid some repetition.

```tsx
type MatomoEvent = {
  eventCategory: string
  eventAction: string
  eventName: string
}

// now you can extend or use this type in other components, instead of repeating the same 3 props every time
interface Props {
  ...
  matomoEvent: MatomoEvent
} 
```

File: src/templates/use-cases.tsx (hunk 2 above)
Comment (corwintines) [reply src/templates/use-cases.tsx]: Pushed up changes for this

**Pull request merged**